### PR TITLE
Migrate gardenlet priority-class

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:
This PR removes `gardenlet` priority class and `gardener-system-critical` and uses `gardener-system-critical-migration` priority class for all the components which are supposed to use `gardener-system-critical` priority class as described in #5634. This is the first step to migrate `gardenlet` priority class to `gardener-system-critical`. 
Once `gardenlet` priority class is cleaned and GRM garbage collect  `gardener-system-critical` priority class, in the next release we can switch `gardener-system-critical-migration` priority class to `gardener-system-critical`.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/5634

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by ary1992** [comment]
Thank you for the PR !
I checked in the local setup, there were no `gardenlet` and `gardener-system-critical` priority-class present. `gardener-system-critical-mig` priority-class is present with value `999998950`.Moreover `gardenlet` deployment is now having `gardener-system-critical-mig` priority class.
/lgtm

File: charts/gardener/gardenlet/charts/runtime/templates/deployment.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ spec:
 {{ toYaml .Values.global.gardenlet.podLabels | indent 8 }}
         {{- end }}
     spec:
-      priorityClassName: gardenlet
+      priorityClassName: gardener-system-critical-mig
```
Comment (shafeeqes) [inline charts/gardener/gardenlet/charts/runtime/templates/deployment.yaml]: why not use the name `gardener-system-critical-migration` fully?

Comment (acumino) [reply charts/gardener/gardenlet/charts/runtime/templates/deployment.yaml]: I am fine with anything but `gardener-system-critical-mig` is smaller.

Comment (timebertt) [reply charts/gardener/gardenlet/charts/runtime/templates/deployment.yaml]: Please use `gardener-system-critical-migration`

**Comment by shafeeqes** [comment]
/lgtm
/approve
Please check the failing tests.

**Review by shafeeqes — APPROVED** [review]

**Pull request merged**